### PR TITLE
fix(aio): replace headers on the temporary marketing home page

### LINKED ISSUE #16824: aio: no anchor tags on the temporary marketing home page.
Sure it's a temporary page but it's annoying and @igorminar wants it gone now.

### PR DESCRIPTION
**Pull request by wardbell**
closes #16824
Although this page is being replaced (#16228), current page has annoying anchor headers. Remove them, replacing h1/h2/h3 with title/section/article and applying the respective header styles to these semantic tags.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: aio/content/marketing/index.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <div class="hero background-superhero-paper is-large">
 
     <img src="assets/images/logos/angular/angular.svg" class="hero-logo"/>
-    <h1 class="text-headline no-toc">One framework.<br>Mobile &amp; desktop.</h1>
+    <title class="text-headline">One framework. Mobile &amp; desktop.</title>
```
Comment (gkalpak) [inline aio/content/marketing/index.html]: No `<br>` inside `<title>` 😞

**Review by gkalpak — APPROVED** [review]
IIRC this has already been discussed and it's been decided to use `title`/`section`/`article` on marketing pages to avoid the link. Given that LGTM

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Sorry I don't know what was agreed but this is a misuse of HTML semantic elements.

For example:

> The section element represents a generic document or application section…The section element is not a generic container element. When an element is needed only for styling purposes or as a convenience for scripting, authors are encouraged to use the div element instead. A general rule is that the section element is appropriate only if the element’s contents would be listed explicitly in the document’s outline.

Further the `title` tag is only valid inside the `head` tag and so should not be used here.

Moreover the generic styling applied in the _typography.scss, making all `title`, `section` and `article` tags match the styling of `h1`, `h2` and `h3` seems wrong. For instance, a section would normally contain prose (including a heading) and would not expect to be styled like a heading; and an `article` is a higher level concept than a `section`.

I believe we should simply use `div` tags to do this styling.

**Comment by sjtrimble** [comment]
Any chance we can hold on this - I have the divs already updated with the appropriate classes assigned and other classes updated.

**Comment by wardbell** [comment]
@petebacondarwin Good point about `<title>`. I see the point. So we should use styled `<div>`. Closing as this isn't the right approach.

**Pull request closed without merging**